app/models/grace_period_submission_rule.rb

```diff
@@ -109,7 +109,7 @@ def remove_deductions(submission)
           deduction.destroy
         end
       end
-      student_membership.user.grace_period_deductions.reload
+      deductions.reload
```

Comment:
No I don't think so.  I must not have done a pull request to get that into my code base while I was testing.  I thought the revised line was better style.